### PR TITLE
Word check added, checking if a word is correct

### PR DESCRIPTION
**Pull request by RogerDMK**
Compares with a game in the database (WordToGuess specifically), currently compares with gameID 1, should be changed to use the current game's ID. Tests have also been added for checkWord. #8

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -1,8 +1,15 @@
-const path = require('path') // used later in the exercise
+'use strict'
```
Comment (JessWhosBack) [inline index.js:1]: Well remembered, we need to check this is in all .js files throughout the project.

File: mainRoutes.js [hunk 2]
```diff
@@ -1,10 +1,62 @@
...
+  req.sendFile(path.join(__dirname, 'views', 'test.html'))
+})
+
+mainRouter.post('/logWord', jsonParser, async function (req, res) {
+  const word = req.body.wordToLog
+
+  if (/^[a-zA-Z]+$/.test(word) === true & word.length === 5) {
+    try {
+      createWord(word).then(data => {
+        // Now we check that the number of rows affected is equal to one since only
```
Comment (JessWhosBack) [inline mainRoutes.js:36]: Good use of comment here!

File: tests/wordQuery.test.js [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+test('Check that the word hello is successfully inserted into the db', () => {
+    const word = 'hello'
+    wordQuery.createWord(word).then(data => {
+        const numRows = JSON.parse(JSON.stringify(data)).rowsAffected.at(0)
+        expect(numRows).toBe(1)
+    })
+})
+
+test('Check that inserting nothing does not insert an element into the db', () => {
```
Comment (JessWhosBack) [inline tests/wordQuery.test.js:11]: Useful edge case test, nice

File: mainRoutes.js [hunk 4]
```diff
@@ -1,10 +1,62 @@
...
+        // one row must be added. If this is anything but one, then an error has occurred.
+        const numRows = JSON.parse(JSON.stringify(data)).rowsAffected.at(0)
+        // console.log(`Num Rows modified: ${numRows}`) for debug
+
+        if (numRows === 1) {
+          checkWord(word, 1).then((check) => {
+            if (check) {// Displays if the guessed word is correct
+              res.send(JSON.stringify({ 'message': `${word} has been saved to the database, you guess the correct word!` }));
```
Comment (JessWhosBack) [inline mainRoutes.js:44]: Possibly bad English phrase - should be past tense of guess

File: mainRoutes.js [hunk 5]
```diff
@@ -1,10 +1,62 @@
...
+          }).catch(console.error)
+        } else {
+          res.send(JSON.stringify({ message: `There was an error saving ${word} to the database` }))
+        }
+      })
+    } catch (err) {
+      console.log(err)
+    }
+  } else {
+    res.send(JSON.stringify({ message: `${word} is invalid. It must be 5 letters long and only be alphabetical` }))
```
Comment (JessWhosBack) [inline mainRoutes.js:58]: Add full stop to sentence

File: public/guessWord.js [hunk 6]
```diff
@@ -0,0 +1,57 @@
+'use strict'
+
+const request = new XMLHttpRequest()
+request.addEventListener('error', onError)
+
+function sendWordToServer () {
+  // Let's get the word that the user input to the textbox (with an id of 'wordID').
```
Comment (JessWhosBack) [inline public/guessWord.js:7]: Great use of comments

File: services/wordQuery.js [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+module.exports = {
+  createWord: async function createWord (word) {
+    return new Promise((resolve, reject) => {
+      const sqlCode = `INSERT INTO [HazardaGuess_db].[dbo].[WordLog] (Word) VALUES ('${word}');`
+      get('default').then(
+        (pool) => pool.request().query(sqlCode).then(
+          (result) => {
+            resolve(result)
+          }
+        )// .catch(reject(console.error)) // TODO: find out why this is causing the server to crash
```
Comment (JessWhosBack) [inline services/wordQuery.js:12]: I can see the to do, but find out why this is causing the server to crash in future

File: services/wordQuery.js [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+    })
+  },
+  checkWord: async function checkWord(word, gameId) {
+    // Gets the word to guess from then checks with the guessed word
+    const sqlCode = `SELECT WordToGuess FROM [dbo].[Game] WHERE GameID='${gameId}';`
+    return new Promise((resolve, reject) => {
+        get('default').then(
+            (pool) => pool.request().query(sqlCode).then(
+                    (result) => {
+                        console.log(result.recordset[0].WordToGuess)
```
Comment (JessWhosBack) [inline services/wordQuery.js:23]: Great use of converting the promise to useable text. Need to hide this console output in the future.

File: tests/wordQuery.test.js [hunk 9]
```diff
@@ -0,0 +1,34 @@
+const wordQuery = require('../services/wordQuery')
+
+test('Check that the word hello is successfully inserted into the db', () => {
```
Comment (JessWhosBack) [inline tests/wordQuery.test.js:3]: Can the user insert an illegal word? (i.e. has weird characters, numbers, etc.)

File: views/test.html [hunk 10]
```diff
@@ -0,0 +1,15 @@
+<!DOCTYPE html>
```
Comment (JessWhosBack) [inline views/test.html:1]: Not a great file name as it is not very descriptive.

**Review by JessWhosBack — APPROVED** [review]
Design: 
- Overall good and consistent design

Functionality: 
- The code does what you intend, and you have tested for some edge cases. However, further edge cases are possible and need to be tested.

Complexity:
- The code is written in a simple enough manner to not be over-engineered. Well done. 

Tests:
- As mentioned above, there is space to expand the tests to include more edge cases.

Naming:
- Great naming convention kept at all times.

Comments:
- Code is well commented.

Style:
- Most style is according to the standardjs notation. Double check for future. 

Consistency:
- Code is very consistent.
